src/util/misc/pick.ts

```diff
@@ -15,3 +15,15 @@ export const pick = <T extends Record<string, any>>(
     return o;
   }, {} as Partial<T>);
 };
+
+export const pickBy = <T extends Record<string, any>>(
+  source: T,
+  predicate: <K extends keyof T>(value: T[K], key: K, collection: T) => boolean
+) => {
+  return (Object.keys(source) as (keyof T)[])
+    .filter((key) => predicate(source[key], key, source))
+    .reduce((o, key) => {
+      o[key] = source[key];
+      return o;
+    }, {} as Partial<T>);
```

Comment:
why the double loop rather than ust usin the if.
```
if(predicate(source[key], key, source)){
  o[key] = source[key];
}
```

Reply:
true